### ISSUE TITLE
testWeightedBestContentTypeForFirefox test failure

### ISSUE THREAD

**Issue by zednis** [tetherless-world/dco-vivo#12]
I get the following error when running tests on my laptop (OSX 10.10.5, Java 8).  Has anyone else seen these errors?  Does anyone know if this is an environment issue?

```
test:
    [mkdir] Created dir: /Users/szednik/Projects/DCO/dco-vivo/.build/main/testClasses
    [javac] Compiling 125 source files to /Users/szednik/Projects/DCO/dco-vivo/.build/main/testClasses
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 1.7
    [javac] Note: Some input files use unchecked or unsafe operations.
    [javac] Note: Recompile with -Xlint:unchecked for details.
    [javac] 1 warning
     [java] Starting test run at 19:50:33 on Aug 16, 2015
     [java] 
     ...
     [java] TEST FAILED: test testWeightedBestContentTypeForFirefox() in edu.cornell.mannlib.vitro.webapp.web.ContentTypeTest: null expected:<[application/xhtml+x]ml> but was:<[text/ht]ml>
     [java]     4    3     0    1     0    0.00 ContentTypeTest
     ...
     [java] 
     [java] Tests Pass Error Fail Ignore Seconds  TOTAL (89 classes)
     [java]   978  950     0    1    27    6.69
     [java] Ending test run at 19:50:39 on Aug 16, 2015
     [java] 

BUILD FAILED
/Users/szednik/Projects/DCO/dco-vivo/vitro/webapp/build.xml:362: Java returned: 1
```

**Comment by zednis** [comment]
I think this might be a Java 8 related issue.

**Comment by zednis** [comment]
Confirmed this is a Java 8 issue. https://jira.duraspace.org/browse/VIVO-766

There is a fix in Vitro 1.8.

The fix is to update the failing test to look like this:

```
     /**
     * Modified this, added q-factor to text/html, because otherwise the result is indeterminate, and the
     * test fails in Java 8.
     */ 
    @Test
    public void testWeightedBestContentTypeForFirefox(){
        //accept header from normal firefox
        Map<String,Float> clientAccepts = ContentType.getTypesAndQ(
        "text/html;q=0.95,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8");

        Map<String,Float> serverTypes = IndividualController.ACCEPTED_CONTENT_TYPES;

        Assert.assertEquals("application/xhtml+xml", ContentType.getBestContentType(clientAccepts, serverTypes)); 
    }
```
